app/src/main/java/com/github/bgabriel998/softwaredevproject/CameraPreview.java

```diff
@@ -187,4 +314,76 @@ private int aspectRatio(int width, int height){
         }
         return new Pair<>((float) Math.toDegrees(horizontalAngle), (float) Math.toDegrees(verticalAngle));
     }
+    
+    /**
+     * Takes a picture of the camera-preview without the canvas drawn
+     */
+    public void takePicture(){
+        //Create the file
+        File photoFile = Button1Activity.createFile(context, FILENAME, PHOTO_EXTENSION);
+
+        //Configure output options
+        ImageCapture.OutputFileOptions outputOptions = new ImageCapture.OutputFileOptions.Builder(
+                photoFile).build();
+
+        //Take the picture
+        imageCapture.takePicture(outputOptions, cameraExecutor, onImageSavedCallback(photoFile));
+    }
+
+    /**
+     * Creates the onImageSavedCallback for when a picture is saved
+     * @param photoFile File that is saved
+     * @return ImageCapture.onImageSavedCallback
+     */
+    private ImageCapture.OnImageSavedCallback onImageSavedCallback(File photoFile){
+        return new ImageCapture.OnImageSavedCallback() {
+            @Override
+            public void onImageSaved(@NonNull ImageCapture.OutputFileResults outputFileResults) {
+                //Get the Uri of the saved picture
+                Uri savedUri = outputFileResults.getSavedUri() != null ? outputFileResults.getSavedUri() : Uri.fromFile(photoFile);
+                handler.sendMessage(handler.obtainMessage(PICTURE_TAKEN, savedUri));
+            }
+
+            @Override
+            public void onError(@NonNull ImageCaptureException exception) {
+                handler.sendMessage(handler.obtainMessage(FAILED_TO_TAKE_PICTURE, exception));
+            }
+
+            //Create a handler to display the toast inside of the onImageSaved callback
+            final Handler handler = pictureMessageHandler();
+        };
+    }
+
+    /**
+     * Handler that is used to display the toast inside the onImageSavedCallback
+     * @return Handler for the toast messages
+     */
+    private Handler pictureMessageHandler(){
+        return new Handler(Looper.getMainLooper()){
+            @Override
+            public void handleMessage(@NonNull Message msg) {
+                super.handleMessage(msg);
+                //Make the toasts depending on the Message code
+                if(msg.what == FAILED_TO_TAKE_PICTURE){
+                    String toastMessage = "Failed to take picture: " + msg.obj.toString();
```

Comment:
Consider adding the string to the `res/string.xml` file for future app translation.
_No suggested change here, the modification implies the editing of two files_

Reply:
Updated string.xml file